apps/cmp.c

```diff
@@ -0,0 +1,3256 @@
+            return 0;
+        }
+    } else {
+        CMP_err("no cmp command to execute");
+        return 0;
+    }
+
+#ifdef OPENSSL_NO_ENGINE
+# define FORMAT_OPTIONS (OPT_FMT_PEMDER | OPT_FMT_PKCS12 | OPT_FMT_ENGINE)
+#else
+# define FORMAT_OPTIONS (OPT_FMT_PEMDER | OPT_FMT_PKCS12)
+#endif
+
+    if (opt_keyform_s != NULL
+            && !opt_format(opt_keyform_s, FORMAT_OPTIONS, &opt_keyform)) {
+        CMP_err("unknown option given for key loading format");
+        return 0;
+    }
+
+#undef FORMAT_OPTIONS
+
+    if (opt_certform_s != NULL
+            && !opt_format(opt_certform_s, OPT_FMT_PEMDER, &opt_certform)) {
+        CMP_err("unknown option given for certificate storing format");
+        return 0;
+    }
+
+    if (opt_certsform_s != NULL
+            && !opt_format(opt_certsform_s, OPT_FMT_PEMDER | OPT_FMT_PKCS12,
+                           &opt_certsform)) {
+        CMP_err("unknown option given for certificate list loading format");
+        return 0;
+    }
+
+    return 1;
+}
+
+static OSSL_CMP_SRV_CTX *setup_srv_ctx(ENGINE *e)
+{
+    OSSL_CMP_CTX *ctx = NULL; /* extra CMP (client) ctx partly used by server */
+    OSSL_CMP_SRV_CTX *srv_ctx = ossl_cmp_mock_srv_new();
+
+    if (srv_ctx == NULL)
+        return NULL;
+    ctx = OSSL_CMP_SRV_CTX_get0_cmp_ctx(srv_ctx);
+
+    if (opt_srv_ref == NULL) {
+        if (opt_srv_cert == NULL) {
+            /* opt_srv_cert should determine the sender */
+            CMP_err("must give -srv_ref for server if no -srv_cert given");
+            goto err;
+        }
+    } else {
+        if (!OSSL_CMP_CTX_set1_referenceValue(ctx, (unsigned char *)opt_srv_ref,
+                                              strlen(opt_srv_ref)))
+            goto err;
+    }
+
+    if (opt_srv_secret != NULL) {
+        int res;
+        char *pass_str = get_passwd(opt_srv_secret, "PBMAC secret of server");
+
+        if (pass_str != NULL) {
+            cleanse(opt_srv_secret);
+            res = OSSL_CMP_CTX_set1_secretValue(ctx, (unsigned char *)pass_str,
+                                                strlen(pass_str));
+            clear_free(pass_str);
+            if (res == 0)
+                goto err;
+        }
+    } else if (opt_srv_cert == NULL) {
+        CMP_err("server credentials must be given if -use_mock_srv or -port is used");
+        goto err;
+    } else {
+        CMP_warn("server will not be able to handle PBM-protected requests since -srv_secret is not given");
+    }
+
+    if (opt_srv_secret == NULL
+            && ((opt_srv_cert == NULL) != (opt_srv_key == NULL))) {
+        CMP_err("must give both -srv_cert and -srv_key options or neither");
+        goto err;
+    }
+    if (opt_srv_cert != NULL) {
+        X509 *srv_cert = load_cert_pwd(opt_srv_cert, opt_srv_keypass,
+                                       "certificate of the server");
+        if (srv_cert == NULL || !OSSL_CMP_CTX_set1_cert(ctx, srv_cert)) {
```

Comment:
nit: blank line after declaration.

Reply:
fixed, also a couple of more such occurrences.